core/src/main/java/org/elasticsearch/search/suggest/SuggestBuilder.java

```diff
@@ -98,6 +98,15 @@ public SuggestBuilder addSuggestion(SuggestionBuilder<?> suggestion) {
         return suggestions;
     }
 
+    /**
+     * Returns the possibly null global suggest text that
+     * should be applied as the text for all suggesters.
+     */
+    @Nullable
+    public String getGlobalText() {
```

Comment:
If you go with this, can you also remove the existing `getText()` and rename the `setText()` to `setGlobalText()`? Also if changing Java API method names, it would be good to keep track of it in `docs/reference/migration/migrate_3_0.asciidoc`, there's a section for Java API changes somewhere. I might have missed some renaming myself in the past, but its good to take a note there if we change anything in the builders that users might directly have in their code.